src/couchapps/WorkQueue/lists/workRestrictions.js

```diff
@@ -13,6 +13,13 @@ function(head, req) {
         return;
     }
 
+    try {
+        var num_elem = JSON.parse(req.query.num_elem);
+    } catch (ex) {
+        send('"Error parsing number of elements" ' +  req.query.num_elem);
```

Comment:
@todor-ivanov to make it sort of backwards compatible, instead of raising an error here, we could set it to a large default value and move on. This way we do not need to backport this feature to the agents. Of course, in a later stage we need to clean this up. What do you think?

Reply:
Just changed my mind, the sooner we get all the agents with this fix in, the safer we are against another avalanche of workflows and/or WQE.